onnx/defs/generator/defs.cc

```diff
@@ -119,6 +119,81 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
+static const char* EyeLike_ver9_doc = R"DOC(
+Generate a 2D tensor (matrix) with ones on the diagonal and zeros everywhere else. Only 2D 
+tensors are supported, i.e. input T1 must be of rank 2.  The shape of the output tensor is the 
+same as the input tensor. The data type can be specified by the 'dtype' argument. If 
+'dtype' is not specified, then the type of input tensor is used.
+
+The 'dtype' argument must be one of the data types specified in the 'DataType' enum field in the
+TensorProto message and be valid as an output type.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    EyeLike,
+    9,
+    OpSchema()
+        .SetDoc(EyeLike_ver9_doc)
+        .Attr(
+            "dtype",
+            "(Optional) The data type for the elements of the output tensor. If not specified,"
+            "the data type of the input tensor T1 is used. If input tensor T1 is also not" 
+            "specified, then type defaults to 'float'.",
+            AttributeProto::INT,
+            OPTIONAL)
+        .Input(
+            0,
+            "input",
```

Comment:
why not have number of rows and columns as parameters rather than infer it from an input tensor? numpy, pytorch and mxnet all have explicit rows and cols as parameters.

Reply:
I think we can have another operator called `Eye` which takes rows and columns as inputs.

Reply:
won't that be like duplicating the functionality? are there examples where this Eyelike operator is used?

Reply:
Yes, that's the idea. The `*Like` ops take the properties such as shape and type from the input tensor. This design is similar to Numpy (`ones_like`, `zero_like`) and also other ops in ONNX such as `ConstantLike`, `RandomNormalLike` and `RandonUniformLike`. 

We can have a separate PR for `Eye` if needed.

Reply:
@anirudhacharya - I don't think this is exactly duplicating functionality. There are several frameworks that subscribe to the design where there are two parallel set of generator ops - one that take tensor as input and one that take explicit shape as input, e.g. (`ones`, `ones_like`), (`zeros`, `zeros_like`). This kind of op is quite useful, e.g. creating epsilon covariance matrices matrices for regularization, and matrix factorization scenarios.

Also, note that I have not added `Eye` in this PR in order avoid adding more ops than necessary, because something like `Eye`, with static shape, can be supported using `Constant` op in current spec.